### ISSUE TITLE
Rust analyzer doesn't show errors

### ISSUE THREAD

**Issue by Squirrelcoding** [rust-analyzer/rust-analyzer#11450]
**rust-analyzer version**: ba3305480 2022-02-07 stable

**rustc version**: 1.58.1

For some reason rust-analyzer stopped showing errors, but when I run `cargo check`, it does show errors. For instance the following piece of simple code doesn't compile, and rust-analyzer says nothing:
```rs
fn main() {
    println!("{}", this_variable_does_not_exist);
}
```
If I hover over `this_variable_does_not_exist`, it just says `{unknown}`. But all of the other features work just fine as far as I know, 

After investigating for a few minutes I realized that in VS code the rust-analyzer thing is grayed out.
![Screenshot (48)](https://user-images.githubusercontent.com/68344468/153533927-fe22c7e7-5f93-4b68-97e5-3cb121c1c201.png)
But how can this be solved?

**Comment by Veykril** [comment]
When you r-a shows the warning symbol it most likely indicates that some cargo command failed to run(usually a build error) which causes us to be unable to fetch diagnostic output from cargo. Does hovering the icon show you any message?

**Comment by Squirrelcoding** [comment]
@Veykril Hovering it over the icon doesn't show anything but I believe I found the cause.

### TL;DR
The problem is not with the extension itself, bur rather a problem with a library which is only supported on Linux. I believe `rust-analyzer` is running on a Windows process as the phenomena only occurs on a VS Code window running on Windows. Running a remote development window in WSL fixes the issue as Rust can find the library on Linux.

---
Checking the output pane for `Rust Analyzer Language Server` shows the following error:
```
thread 'main' panicked at 'compilation failed: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }', C:\Users\...\.cargo\registry\src\github.com-1ecc6299db9ec823\ncurses-5.101.0\build.rs:105:30
```
I am using the `ncurses` library which is used for building TUIs *and* is only found on Linux. The thing is that `rust-analyzer` can't find it and is causing the error, when running `rust-analyzer` on a remote development window in WSL, everything appears to work just fine as Rust can find the library on there. So the problem isn't with the extension itself exactly, but rather a problem between the compatibility between Linux and Windows. So the only solution (as far as I know) is to use `rust-analyzer` in a remote development window.

**Comment by Insem** [comment]
Hi, I have the same problem but I use manjaro linux and have installed the `ncurses` lib.

https://archlinux.org/packages/core/x86_64/ncurses/

![image](https://user-images.githubusercontent.com/20770996/185902188-9fc6599b-162c-415c-900e-8836099875ef.png)

**Comment by lnicola** [comment]
@Insem can you try adding this to your Code settings?

```json
    "editor.semanticTokenColorCustomizations": {
        "rules": {
            "unresolvedReference": "#ff0000",
        },
    }
```
